demo/server/src/api/meta.js

```diff
@@ -0,0 +1,50 @@
+const https = require('https');
+const aws4 = require('aws4');
+
+export const handler = (event, context, callback) => {
+  const region = process.env.REGION;
+  const apiGatewayId = process.env.META_INDEX_API_ID;
+  const stage = process.env.STAGE;
+  const endpoint = '/meta-index/list';
+  const accessKeyId = process.env.PRODUCER_KEY_ID;
+  const secretAccessKey = process.env.PRODUCER_SECRET_ACCESS_KEY;
+
+  const params = {
+    host: `${apiGatewayId}.execute-api.${region}.amazonaws.com`,
+    method: 'GET',
+    path: `/${stage}${endpoint}`,
+  };
+
+  // can specify any custom option or header as per usual
+  const req = https.request(
+    aws4.sign(params, {
+      accessKeyId,
+      secretAccessKey,
+    }),
+    res => {
+      res.setEncoding('utf8');
+      let body = '';
+
+      res.on('data', data => {
+        body += data;
```

Comment:
So far I don't see any obvious reason this operation feels a bit slower, the only concept coming to mind is streams. Not sure whether piping data to `res` will do any good compared to aggregating the chunks and returning the result at once.

Reply:
I don't see your point. This is the common way to make requests with Node.js